discretisedfield/mesh.py

```diff
@@ -187,22 +182,42 @@ def __init__(
         attributes={"unit": "m"},
     ):
         # TODO NO MUTABLE DEFAULT
-        if region is not None and p1 is None and p2 is None:
-            self.region = region
+        if region is None and p1 is None and p2 is None:
+            raise ValueError("p1, p2, and region cannot be None simultaneously.")
+        elif region is not None and p1 is None and p2 is None:
+            if not isinstance(region, df.Region):
+                raise TypeError("region must be of class discretisedfield.Region.")
+            self._region = region
         elif region is None and p1 is not None and p2 is not None:
-            self.region = df.Region(p1=p1, p2=p2)
+            self._region = df.Region(p1=p1, p2=p2)
         else:
             msg = "Either region or p1 and p2 can be passed, not both."
             raise ValueError(msg)
 
-        if cell is not None and n is None:
-            self.cell = tuple(cell)
-            n = np.divide(self.region.edges, self.cell).round().astype(int)
-            self.n = dfu.array2tuple(n)
+        if cell is None and n is None:
+            raise ValueError("Both cell and n cannot be None.")
+        elif cell is not None and n is None:
+            if not isinstance(cell, Iterable):
+                raise TypeError("Cell must be an Iterable.")
```

Comment:
I think `Iterable` is too generic. I would only allow array-like types (tuple, list, np.ndarray). I am not sure if there is a generic test for this.

Reply:
The value of `cell` is a genexp in [`io.field_from_ovf`](https://github.com/ubermag/discretisedfield/blob/c8ff5594418189801d1f750e57e22a32e239e09a/discretisedfield/io/ovf.py#L246), and hence, raises `TypeError`. If you like I can change this in the `io` module.

Reply:
That would be good.